applications/file-split-ftp/src/main/java/org/springframework/integration/samples/filesplit/Application.java

```diff
@@ -0,0 +1,238 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.samples.filesplit;
+
+import java.io.File;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+
+import org.aopalliance.intercept.MethodInterceptor;
+import org.apache.commons.net.ftp.FTPFile;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.autoconfigure.mail.MailProperties;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.integration.dsl.IntegrationFlow;
+import org.springframework.integration.dsl.IntegrationFlows;
+import org.springframework.integration.dsl.core.Pollers;
+import org.springframework.integration.dsl.file.Files;
+import org.springframework.integration.dsl.mail.Mail;
+import org.springframework.integration.dsl.support.StringStringMapBuilder;
+import org.springframework.integration.file.FileHeaders;
+import org.springframework.integration.file.remote.session.SessionFactory;
+import org.springframework.integration.file.splitter.FileSplitter;
+import org.springframework.integration.file.support.FileExistsMode;
+import org.springframework.integration.ftp.session.DefaultFtpSessionFactory;
+import org.springframework.integration.http.config.EnableIntegrationGraphController;
+import org.springframework.integration.router.PayloadTypeRouter;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageHeaders;
+import org.springframework.messaging.MessagingException;
+import org.springframework.web.servlet.config.annotation.CorsRegistry;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurerAdapter;
+
+@SpringBootApplication
+@EnableConfigurationProperties(MailProperties.class)
+@EnableIntegrationGraphController
+public class Application {
+
+	private static final String EMAIL_SUCCESS_SUFFIX = "emailSuccessSuffix";
+
+	public static void main(String[] args) {
+		SpringApplication.run(Application.class, args);
+	}
+
+	@Autowired
+	private MailProperties mailProperties;
+
+	@Bean
+	public IntegrationFlow fromFile() {
+		return IntegrationFlows.from(Files.inboundAdapter(new File("/tmp/in"))
+					.patternFilter("*.txt"), e -> e.poller(Pollers.fixedDelay(5000)))
+				.enrichHeaders(h -> h.header(MessageHeaders.ERROR_CHANNEL, "tfrErrors.input"))
```

Comment:
https://jira.spring.io/browse/INT-4113.
I mean add comment here.
Although I think after `getComponentsToRegister()` for the `PollerSpec` we can provide some out-of-the-box workaround just with `errorChannel(MessageChannelString)`
But right: that is a different story...